includes/polling/cisco-nac.inc.php

```diff
@@ -13,11 +13,43 @@
         $port_auth_id_nac = strstr($PortAuthSessionEntryNAC, "'");
         $port_auth_id_nac = substr($port_auth_id_nac, 0, -1);
         $port_auth_id_nac = substr($port_auth_id_nac, 1);
-        $CheckExistIndex = dbFetchRow('SELECT * FROM `ports_nac` WHERE `port_index` = ?', $port_index_nac);
-        $CheckExistAuth = dbFetchRow('SELECT * FROM `ports_nac` WHERE `auth_id` = ?', $port_auth_id_nac);
+        $port_auth_domain = $PortAuthSessionEntryParameters['cafSessionDomain'];
         $IPHextoDec = IP::fromHexString($PortAuthSessionEntryParameters['cafSessionClientAddress']);
-        if ($CheckExistIndex['port_index'] != null) {
-            echo "Port Index Was Found. Updating...\n";
+        $CheckExistIndex = dbFetchRows('SELECT port_index FROM `ports_nac` WHERE `port_index` = ?', $port_index_nac);
```

Comment:
Instead of dbFetchRows(), you can use dbFetchRow() to selected a single row (that way you can avoid the [0] below)

Even better, if you are selecting a single item, you can dbFetchCell() to select a single value from the db and it won't be in an array.

Reply:
Done. It make all sense. Thanks for advising.